src/web/utils/FocusManager.ts

```diff
@@ -157,10 +159,13 @@ export class FocusManager extends FocusManagerBase {
             // In order to avoid losing this first Tab press, we're making <body>
             // focusable, focusing it, removing the focus and making it unfocusable
             // back again.
-            const prevTabIndex = FocusManager._setTabIndex(document.body, 0);
-            document.body.focus();
-            document.body.blur();
-            FocusManager._setTabIndex(document.body, prevTabIndex);
+            // Defer the work to avoid triggering sync layout.
+            FocusManager._resetFocusTimer = setTimeout(() => {
+                const prevTabIndex = FocusManager._setTabIndex(document.body, 0);
+                document.body.focus();
```

Comment:
Please have Marat review this to make sure it's not introducing an undesirable race condition. For example, what if the user presses the tab key to move the focus to the next item? This key event used to arrive after the synchronous code ran, but now it can arrive before the deferred code runs.

Reply:
You're absolutely right, it broke focusing even without an explicit user action. Fixed in 66bfc976b by making the deferred code really idempotent. CC @mshoho